src/hooks/usePairLiquidityMiningCampaigns.ts

```diff
@@ -0,0 +1,163 @@
+import { useCallback, useMemo } from 'react'
+import { gql, useQuery } from '@apollo/client'
+import { BigintIsh, Pair } from '@swapr/sdk'
+import { SubgraphLiquidityMiningCampaign } from '../apollo'
+
+import { useNativeCurrency } from './useNativeCurrency'
+import { useActiveWeb3React } from '.'
+import {
+  getLowerTimeLimit,
+  getTokenAmount,
+  sortActiveCampaigns,
+  sortExpiredCampaigns,
+  toLiquidityMiningCampaign,
+} from '../utils/liquidityMining'
+import { useAllTokensFromActiveListsOnCurrentChain } from '../state/lists/hooks'
+import { useKpiTokens } from './useKpiTokens'
+
+const CAMPAIGN_REWARDS_TOKEN_COMMON_FIEDLDS = ['address: id', 'name', 'symbol', 'decimals', 'derivedNativeCurrency']
+const CAMPAIGN_COMMON_FIEDLDS = ['duration', 'startsAt', 'endsAt', 'locked', 'stakingCap', 'stakedAmount']
```

Comment:
Not sure if this is best approach since it only used in this one file but it would make more sense if we would to reuse some common query patterns across all the liqudity mining campaings....with difference added just in file its used.